design-proposals/virt-operator-deploy-instancetypes.md

```diff
@@ -0,0 +1,97 @@
+[specific release](https://github.com/kubevirt/common-instancetypes/releases/tag/v0.3.1)
+or by using the [SSP operator](https://github.com/kubevirt/ssp-operator#ssp-operator).
+
+This proposal suggests moving the deployment of the cluster-wide
+`common-instancetypes` resources to `virt-operator` so that they are
+consistently available across KubeVirt deployments.
+
+## Motivation
+
+The ultimate goal of this proposal is to improve KubeVirt's default user
+experience when creating `VirtualMachines` across several different deployments.
+
+The deployment by default of the cluster-wide `common-instancetypes` resources
+will allow users to rely more freely on the presence of the resources and make
+use of them during their `VirtualMachine` creation flows.
+
+## Goals
+
+* For new deployments, the deployment of the cluster-wide `common-instancetypes`
+  resources by `virt-operator`
+
+* For upgraded deployments, to have `virt-operator` claim ownership of any
+  existing cluster-wide `common-instancetypes` resources deployed by the SSP
+  operator.
+
+## Non Goals
+
+* For upgraded deployments, `virt-operator` will not try to claim ownership of
+  any operator deployed cluster-wide `common-instancetypes` resources.
+
+## User Stories
+
+* As a user I want access to a consistent default set of cluster-wide instance
+  types and preferences across all deployments I interact with
+
+* As an operator I want my users to have access to a consistent set of
+  cluster-wide instance types and preferences without the need for additional
+  deployment of operators or resources
+
+## Repos
+
+* kubevirt/kubevirt
+* kubevirt/common-instancetypes
+* kubevirt/ssp-operator
+
+# Design
+
+**TODO** Add more context here once some draft code has been written
+
+`virt-operator` will be extended to by default deploy bundles of cluster-wide
+resources from the `common-instancetypes` project.
+
+**TODO** Should we also support deploying resources from a kustomize URI as is
+         supported by SSP Operator?
+
+These bundles will be sync'd into the KubeVirt tree during the release of a
+given `common-instancetypes` version as currently happens with the SSP operator.
+
+## Scalability
+
+The `common-instancetypes` project currently ships instance type and preference
+cluster-wide resources. Their deployment by default should not introduce any
+scalability issues.
+
+## Update/Rollback Compatibility
+
+For upgraded KubeVirt deployments `virt-operator` will attempt to take ownership
+of existing `common-instancetypes` cluster-wide resources previously deployed by
+the SSP operator. The SSP operator itself will need to be updated at the same
+time to ensure it relinquishes ownership and no longer attempts to reconcile the
+resources.
+
+Rolling both `virt-operator` and SSP operator back to their previous versions
+should return control of these resources to the SSP operator.
+
+## Functional Testing Approach
+
+TBD
```

Comment:
Unit tests for all prefs+its
One e2e test per pref.
Pref is more sensible than ITs, thus I do not think that we have to test the full pref vs IT matrix.

Reply:
As I said before merging even a subset of the `common-instancetypes` tree into `kubevirt/kubevirt` migrates the maintenance and testing burden into an already busy repo. Using a submodule approach would allow us to focus on the specific fresh deployment and upgrade scenarios here.